src/scicloj/02_numpy.clj

```diff
@@ -439,17 +858,59 @@ plt.ylabel('number'));
 ^k/dataset
 (ds/head csv-data 3)
 
+(tech.v3.datatype.functional/mean (dtype/->reader (-> csv-data last last)))
+
+(letfn [(mean [x] (float (/ (apply + x) (count x))))]
+  (nd-aggregate mean
+                (dtype/->reader (-> csv-data last last))))
+
+
 ["Computation on Arrays: Broadcasting
  ------------------------------------------------"]
 
-["### WIP"]
-
 ["Comparisons, Masks, and Boolean Logic
  ------------------------------------------------"]
 
+(defn mask [f t]
+  (-> (dtype/emap #(if (f %) 1 0) :int32 t)
+      dtt/->tensor
+      ravel))
+
```

Comment:
I guess indexed-buffer could be the answer: use a tensor with a boolean operator to somehow produce the index of the truthy values, then use that to create a derived indexed-buffer for use in computations.

Reply:
I agree, there's probably a better way to do this.